src/realm/object-store/c_api/config.cpp

```diff
@@ -34,8 +39,10 @@ RLM_API bool realm_config_set_encryption_key(realm_config_t* config, const uint8
             throw InvalidEncryptionKey();
         }
 
-        config->encryption_key.clear();
-        std::copy(key, key + key_size, std::back_inserter(config->encryption_key));
+        std::array<uint8_t, 64> raw_key;
+        std::copy(key, key + key_size, raw_key.begin());
+        config->encryption_key.emplace(raw_key);
```

Comment:
Initializing encryption keys from buffers seems to be done in a different way every time, and almost all of them involve several steps. The EncryptionKey type should just have a constructor which takes a buffer.

Reply:
That is a weakness of the API, yeah. I’m not that well-versed in C++ template specialization and I couldn’t find a way to specialize SensitiveBuffer<std::array<uint8_t, 64>> that adds a constructor accepting a char* and copies that into the internal buffer. But now that we have a header just for the EncryptionKeyType alias perhaps we could also add a helper function that refactors all the array copy shenanigans and implements it more cleanly?
Alternatively one can argue there’s no point to a generic SensitiveBuffer<> and just collapse all this into an EncryptionKey class and not deal with the complexity we have now.

Reply:
I don't think it makes sense to have a generic type that's used for exactly one thing if it's not even a good fit for that one thing.

Reply:
That’s very fair. @kiburtse how do you feel about taking a stab at this?

Reply:
sure, just to be clear, we are talking here about simply scrapping SensitiveBuffer and implementing all the logic in simple separate class EncryptionKey with all needed constructors and methods to get rig off this unneeded copy?

Reply:
I believe that’s the way forward, yeah.